modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/resources/manifest/ExportedArtifactsCollector.java

```diff
@@ -198,6 +201,23 @@ public void visitObject(ObjectType objectType)
                 {
                     accept.set(true);
                 }
+
+                @Override
+                public void visitString(StringType stringType)
+                {
+                    accept.set(stringType.getAnnotation(TypeIdAnnotation.class).stream().anyMatch(
```

Comment:
MetadataUtils class has a method which does this and returns the class associated with the type. That would make for a clerarer implementation of this method